ex03.ex

```diff
@@ -59,8 +59,20 @@ defmodule Ex03 do
 
   """
 
-  def odd_even . . . "your code"
+  def map([], _func), do: []
+  def map([ h | t ], func), do: [ func.(h) | map(t, func) ]
 
```

Comment:
# +1

Normally, I say not to use conditional statements in code, simply because it's a basic rule that's easy to follow when you're just starting out.

However, I see from this code that you're past that stage, and you showed good judgement: the `map` abstraction was a nice touch, and the use of `cond` is probably justified.

My only suggestion: there's no need for both `Integer.is_odd` and `Integer.is_even`. The first test guarantees that `n` is an integer, so the second could just be `true -> :even`.